File: camerax_android/lib/src/legacy/video/fallback_strategy.dart
import 'package:camerax_android/src/legacy/camerax.g.dart' as $native;
import 'package:camerax_platform_interface/camerax_platform_interface.dart'
    as $interface;

import 'quality.dart';

final class FallbackStrategy extends $interface.FallbackStrategy {
  final $native.FallbackStrategy obj;

  FallbackStrategy.$native(this.obj) : super.impl();

  factory FallbackStrategy.higherQualityOrLowerThan(
      $interface.Quality quality) {
    final obj = $native.FallbackStrategy.higherQualityOrLowerThan(
      quality: quality.obj,
    );
    return FallbackStrategy.$native(obj);
  }

  factory FallbackStrategy.higherQualityThan($interface.Quality quality) {
    final obj = $native.FallbackStrategy.higherQualityThan(
      quality: quality.obj,
    );
    return FallbackStrategy.$native(obj);
  }

  factory FallbackStrategy.lowerQualityOrHigherThan(
      $interface.Quality quality) {
    final obj = $native.FallbackStrategy.lowerQualityOrHigherThan(
      quality: quality.obj,
    );
    return FallbackStrategy.$native(obj);
  }

  factory FallbackStrategy.lowerQualityThan($interface.Quality quality) {
    final obj = $native.FallbackStrategy.lowerQualityThan(
      quality: quality.obj,
    );
    return FallbackStrategy.$native(obj);
  }
}

File: camerax_android/android/src/main/kotlin/dev/hebei/camerax_android/legacy/video/FallbackStrategy.kt
package dev.hebei.camerax_android.legacy.video

import dev.hebei.camerax_android.legacy.CameraXRegistrar
import dev.hebei.camerax_android.legacy.PigeonApiFallbackStrategy

class FallbackStrategy(registrar: CameraXRegistrar) : PigeonApiFallbackStrategy(registrar) {
    override fun higherQualityOrLowerThan(quality: androidx.camera.video.Quality): androidx.camera.video.FallbackStrategy {
        return androidx.camera.video.FallbackStrategy.higherQualityOrLowerThan(quality)
    }

    override fun higherQualityThan(quality: androidx.camera.video.Quality): androidx.camera.video.FallbackStrategy {
        return androidx.camera.video.FallbackStrategy.higherQualityThan(quality)
    }

    override fun lowerQualityOrHigherThan(quality: androidx.camera.video.Quality): androidx.camera.video.FallbackStrategy {
        return androidx.camera.video.FallbackStrategy.lowerQualityOrHigherThan(quality)
    }

    override fun lowerQualityThan(quality: androidx.camera.video.Quality): androidx.camera.video.FallbackStrategy {
        return androidx.camera.video.FallbackStrategy.lowerQualityThan(quality)
    }
}